arangod/Aql/IndexExecutor.cpp

```diff
@@ -62,8 +62,30 @@ using namespace arangodb;
 using namespace arangodb::aql;
 
 namespace {
+
+bool getHasMultipleExpansions(
+    std::vector<transaction::Methods::IndexHandle> const& indexes) noexcept {
+  // count how many attributes in the index are expanded (array index)
+  // if more than a single attribute, we always need to deduplicate the
+  // result later on
+  for (auto const& idx : indexes) {
+    size_t expansions = 0;
+    auto const& fields = idx->fields();
+    for (size_t i = 0; i < fields.size(); ++i) {
+      if (idx->isAttributeExpanded(i)) {
+        ++expansions;
+        if (expansions > 1 || i > 0) {
```

Comment:
Why this `i > 0` check? If `expansions > 1`, then we must have looked at at least two attributes, so the `i > 0` check will always be fulfilled.

Reply:
I simply moved the code snippet from the constructor body to a function in the anonymous namespace, simply to make it callable from the initializer list. The existing logic or the code comment I didn't modify. I will now add a code comment to make it more clear.